nx/lib/nx.ex

```diff
@@ -5244,6 +5244,38 @@ defmodule Nx do
     impl!(tensor).conjugate(%{tensor | type: Nx.Type.to_complex(tensor.type)}, tensor)
   end
 
+  @doc """
+  Calculates the complex phase angle of each element in the tensor.
+
+  If $z = a + bi = r e^\\theta$, $phase(z) = {\rm atan2}(b, a)$
+
+  ## Examples
+
+       iex> Nx.phase(Complex.new(1, 2))
+       #Nx.Tensor<
+         c64
+         1.1071487665176392+0.0i
+       >
+
+       iex> Nx.phase(1)
+       #Nx.Tensor<
+         f32
+         0.0
+       >
+
+       iex> Nx.phase(Nx.tensor([Complex.new(1, 2), Complex.new(-2, 1)]))
+       #Nx.Tensor<
+         c64[2]
+         [1.1071487665176392+0.0i, 2.677945137023926+0.0i]
+       >
+  """
```

Comment:
I feel like this should be a float and not a complex in all cases. Should I add a `Nx.Type.to_float` and set the output tensor type to float?

Reply:
What does EXLA/Numpy return here? It makes me think perhaps we should add support for complex numbers in EXLA before moving forward with adding more functions.

Reply:
I think we should return the corresponding float here c64 -> f32, c128 -> f64, otherwise f32

This would go for abs/real/imag as well. I imagine them as "taking the components apart", so it makes sense to decompose into the underlying datatype instead of keeping them as a complex with no imaginary part

Reply:
> What does EXLA/Numpy return here? 

From what I understand from the docs, np.angle returns a float or an integer.

> It makes me think perhaps we should add support for complex numbers in EXLA before moving forward with adding more functions.

I think it doesn't hurt to add a few more functions. Maybe we could mention in the docs that these functions do not work with EXLA and Torchx?